iib/workers/tasks/fbc_utils.py

```diff
@@ -69,3 +69,29 @@ def get_hidden_index_database(from_index: str, base_dir: str) -> str:
     os.makedirs(os.path.dirname(base_db_file), exist_ok=True)
     _copy_files_from_image(from_index, conf['hidden_index_db_path'], base_db_file)
     return base_db_file
+
+
+def extract_fbc_fragment(temp_dir: str, fbc_fragment: str) -> Tuple:
+    """
+    Extract operator from the from fbc_fragment image.
+
+    :param str temp_dir: base temp directory for IIB request
+    :pararm str fbc_fragment: fbc_fragment in the IIB request
+    :return fbc_fragment path, fbc_operator_package
+    :rtype: tuple
+    """
+    from iib.workers.tasks.build import _copy_files_from_image
+
+    log.info("Extracting the fbc_fragment's catalog from  %s", fbc_fragment)
+    # store the fbc_fragment at /tmp/iib-**/fbc-fragment
+    fbc_fragment_path = os.path.join(temp_dir, 'fbc_fragment')
+    os.makedirs(os.path.dirname(fbc_fragment_path), exist_ok=True)
+
+    conf = get_worker_config()
+    _copy_files_from_image(fbc_fragment, conf['fbc_fragment_catalog_path'], fbc_fragment_path)
+
+    operator_packages = os.listdir(fbc_fragment_path)
```

Comment:
Have you tried this one? I am not sure if `_copy_files_from_image` does not copy whole directory so it means it would list only `catalog` folder. But I might be wrong here.

Reply:
the fbc_fragment_path looks like /tmp/iib-1mnpn1yl/fbc_fragment, this has the content of catalog in fbc_fragment,
os.listdir on fbc_fragment_path displays operator_packages like,  ['example-operator']